src/ng/animate.js

```diff
@@ -416,8 +416,8 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * @param {Promise} animationPromise The animation promise that is returned when an animation is started.
        */
       cancel: function(runner) {
-        if (runner.end) {
-          runner.end();
+        if (runner.cancel) {
+          runner.cancel();
```

Comment:
Hm, we'd have to create a scenario where this can happen. I mean, if the expression changes multiple times during the digest then the watchActionFn is still only called with the result, so there should be no difference. Can you create a real-world example where this will happen?
And I don't think ngInclude, ngView, ngIf are affected because they only ever switch one element with another element.